src/main/java/bookshop/security/JwtUtil.java

```diff
@@ -0,0 +1,63 @@
+package bookshop.security;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import java.util.Date;
+import java.util.function.Function;
+import javax.crypto.SecretKey;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtUtil {
+    private final SecretKey secret;
+
+    @Value("${jwt.expiration}")
+    private long expiration;
+
+    public JwtUtil(@Value("${jwt.secret}") final String secretString) {
+        secret = Keys.hmacShaKeyFor(secretString.getBytes(UTF_8));
+    }
+
+    public String generateToken(final String username) {
+        return Jwts.builder()
+                .subject(username)
+                .issuedAt(new Date(System.currentTimeMillis()))
+                .expiration(new Date(System.currentTimeMillis() + expiration))
+                .signWith(secret)
+                .compact();
+    }
+
+    public boolean isValidToken(final String token) {
+        try {
+            final Jws<Claims> claimsJws = Jwts.parser()
+                    .verifyWith(secret)
+                    .build()
+                    .parseSignedClaims(token);
+            return !claimsJws.getPayload().getExpiration().before(new Date());
+        } catch (final JwtException | IllegalArgumentException exception) {
+            throw new JwtException("Expired or invalid JWT token");
```

Comment:
It's a bad idea to throw and error here, because if token is expired you more likely will get 500 response because of the error. It's better to simply return `false` here.